### PR TITLE
[FLINK-28146][runtime] Sync blocklist information between JobMaster & ResourceManager

### PR DESCRIPTION
**Pull request by wanglijie95**
## What is the purpose of the change

The newly added/updated blocked nodes should be synchronized between JM and RM.

## Verifying this change
`JobMasterTest#testNewlyAddedBlockedNodesWillBeSynchronizedToResourceManager`
`ResourceManagerTest#testNewlyAddedBlockedNodesWillBeSynchronizedToAllRegisteredJobMasters`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (**no**)
  - If yes, how is the feature documented? (**not applicable**)

### CODE REVIEW

**Review by zhuzhurk — APPROVED** [review]
Thanks for opening this PR! @wanglijie95 
The change of the syncing part looks good to me.
Will merge it once #20218 is accepted and merged.

**Comment by zhuzhurk** [comment]
@flinkbot run azure

**Comment by wanglijie95** [comment]
@flinkbot run azure

**Comment by zhuzhurk** [comment]
Merging.

**Pull request closed without merging**
